simulation.cpp

```diff
@@ -396,31 +373,36 @@ void test_simulation() noexcept//!OCLINT test may be many
         identity_env.env_function_A = identity;
         assert(are_same_env_functions(identity_env.env_function_A, identity));
 
-        auto identity_net = net_param{{1,1}, linear};
+        auto identity_net_param = net_param{{1,1}, linear};
+        network identity_net {identity_net_param};
+        identity_net = change_all_weights(identity_net, 1);
+
         assert(net_behaves_like_the_function(identity_net, identity));
 
-        auto identity_ind = ind_param{identity_net};
+        auto identity_ind = ind_param{identity_net_param};
 
         int pop_size = 2;
         auto minimal_pop = pop_param{pop_size, 0, 0};
 
+        auto sim_p = sim_param{};
+        sim_p.selection_strength = 2;
+
 
         simulation s{all_params{
                 identity_env,
                 identity_ind,
                 minimal_pop,
-                sim_param{}
+                sim_p
             }};
 
         //change target value to match output of ind 0 net
         size_t best_ind = 0;
-        auto best_net = get_nth_ind_net(s, best_ind);
-        auto resp_best = response(get_nth_ind(s, best_ind))[0];
+        auto best_net = change_all_weights(get_nth_ind_net(s, best_ind), 1);
+        change_nth_ind_net(s, best_ind, best_net);
 
         size_t worst_ind = 1;
         auto worst_net = change_all_weights(get_nth_ind_net(s,worst_ind), 100);
         change_nth_ind_net(s, worst_ind, worst_net);
```

Comment:
In theory, now that I think about it, this does not required to be changed to not behave like an identity network.
```suggestion
        auto worst_net = get_nth_ind_net(s,worst_ind);
```